nixos/tests/installed-tests/default.nix

```diff
@@ -67,7 +67,7 @@ let
             '' +
             ''
               machine.succeed(
-                  "gnome-desktop-testing-runner ${testRunnerFlags} -d '${tested.installedTests}/share'"
+                  "gnome-desktop-testing-runner ${toString testRunnerFlags} -d '${tested.installedTests}/share'"
```

Comment:
Perhaps use `lib.escapeShellArgs` then?

Reply:
it's already escaped in the execute function of the tester https://github.com/NixOS/nixpkgs/blob/5d533d4aafa1668442ceb1c2a475b37536647213/nixos/lib/test-driver/test_driver/machine.py#L539

Reply:
That just ensures that the whole command is passed as a single argument to `sh`. But if we accept list of arguments, I would expect them to properly handle arbitrary values, unlike a argument string, where the user is expected to ensure individual arguments are properly escaped (c.f. Python’s [`subprocess.run` with and without `shell=True`](https://docs.python.org/3/library/subprocess.html#security-considerations)). Though annoyingly, e.g. `makeFlags` attribute currently does not follow this and values containing spaces will just break.

Reply:
switched